consensus/hotstuff/voteaggregator/vote_aggregator_v2.go

```diff
@@ -63,48 +65,43 @@ func NewVoteAggregatorV2(
 		queuedVotesNotifier: engine.NewNotifier(),
 	}
 
-	return aggregator, nil
-}
+	componentBuilder := component.NewComponentManagerBuilder()
 
-// Ready returns a ready channel that is closed once the engine has fully
-// started. For the propagation engine, we consider the engine up and running
-// upon initialization.
-func (va *VoteAggregatorV2) Ready() <-chan struct{} {
-	va.lm.OnStart(func() {
-		// launch as many workers as we need
-		for i := 0; i < defaultVoteAggregatorWorkers; i++ {
-			va.unit.Launch(va.queuedVotesProcessingLoop)
-		}
-	})
-	return va.lm.Started()
-}
+	// launch as many workers as we need
+	for i := 0; i < defaultVoteAggregatorWorkers; i++ {
+		componentBuilder.AddWorker(func(ctx irrecoverable.SignalerContext, ready component.ReadyFunc) {
+			ready()
+			err := aggregator.queuedVotesProcessingLoop(ctx)
+			if err != nil {
+				ctx.Throw(err)
+			}
+		})
+	}
+
+	aggregator.ComponentManager = componentBuilder.Build()
 
-func (va *VoteAggregatorV2) Done() <-chan struct{} {
-	va.lm.OnStop(func() {
-		<-va.unit.Done()
-	})
-	return va.lm.Stopped()
+	return aggregator, nil
 }
 
-func (va *VoteAggregatorV2) queuedVotesProcessingLoop() {
+func (va *VoteAggregatorV2) queuedVotesProcessingLoop(ctx context.Context) error {
 	notifier := va.queuedVotesNotifier.Channel()
 	for {
 		select {
-		case <-va.unit.Quit():
-			return
+		case <-ctx.Done():
```

Comment:
## ⚠️  Caution with substituting an `irrecoverable.SignalerContext` as `context.Context` 

you **cannot use `SignalerContext.Done()`** as a trigger when to terminate the business logic:
* The `SignalerContext` has the method `ShutdownSignal` to retrieve the shutdown signal. In contrast, the channel from `SignalerContext.Done()` is only closed after _all workers_ terminated. 
* If each individual worker keeps going until all worker terminated (i.e. using `ctx.Done()`), then we can't shutdown the workers.